pkg/diff/sql_generator.go

```diff
@@ -1444,14 +1444,29 @@ type checkConstraintSQLGenerator struct {
 }
 
 func (csg *checkConstraintSQLGenerator) Add(con schema.CheckConstraint) ([]Statement, error) {
-	var hazards []MigrationHazard
-
 	// UDF's in check constraints are a bad idea. Check constraints are not re-validated
 	// if the UDF changes, so it's not really a safe practice. We won't support it for now
 	if len(con.DependsOnFunctions) > 0 {
 		return nil, fmt.Errorf("check constraints that depend on UDFs: %w", ErrNotImplemented)
 	}
 
+	var stmts []Statement
+	if !con.IsValid || csg.isNewTable {
```

Comment:
If `!con.IsValid` shouldn't we have a validateConstraintStatement?

Reply:
Nope! Because the user is adding an invalid constraint! (They want it to be invalid)